controllers/extensionRequests.js

```diff
@@ -14,73 +14,166 @@ const { parseQueryParams } = require("../utils/queryParser");
+      let assigneeId = extensionBody.assignee;
+      if (!assigneeUsername) {
+        assigneeId = await getUserIdElseUndefined(extensionBody.assignee);
+        assigneeUsername = extensionBody.assignee;
+        extensionBody.assignee = assigneeId;
+      }
 
-    if (req.userData.id !== extensionBody.assignee && !req.userData.roles?.super_user) {
-      return res.boom.forbidden("Only assigned user and super user can create an extension request for this task.");
-    }
+      if (!assigneeId) {
+        return res.boom.badRequest("User with this id or username doesn't exist.");
+      }
 
-    const { taskData: task } = await tasks.fetchTask(extensionBody.taskId);
-    if (!task) {
-      return res.boom.badRequest("Task with this id or taskid doesn't exist.");
-    }
-    if (task.assignee !== assigneeUsername) {
-      return res.boom.badRequest("This task is assigned to some different user");
-    }
-    if (task.endsOn >= extensionBody.newEndsOn) {
-      return res.boom.badRequest("The value for newEndsOn should be greater than the previous ETA");
-    }
-    if (extensionBody.oldEndsOn !== task.endsOn) {
-      extensionBody.oldEndsOn = task.endsOn;
-    }
+      if (req.userData.id !== extensionBody.assignee && !req.userData.roles?.super_user) {
+        return res.boom.forbidden("Only assigned user and super user can create an extension request for this task.");
+      }
 
-    const prevExtensionRequest = await extensionRequestsQuery.fetchExtensionRequests({
-      taskId: extensionBody.taskId,
-      assignee: extensionBody.assignee,
-    });
-    if (prevExtensionRequest.length) {
-      return res.boom.forbidden("An extension request for this task already exists.");
+      const { taskData: task } = await tasks.fetchTask(extensionBody.taskId);
+      if (!task) {
+        return res.boom.badRequest("Task with this id or taskid doesn't exist.");
+      }
+      if (task.assignee !== assigneeUsername) {
+        return res.boom.badRequest("This task is assigned to some different user");
+      }
+      if (task.endsOn >= extensionBody.newEndsOn) {
+        return res.boom.badRequest("The value for newEndsOn should be greater than the previous ETA");
+      }
+      if (extensionBody.oldEndsOn !== task.endsOn) {
+        extensionBody.oldEndsOn = task.endsOn;
+      }
+
+      const latestExtensionRequest = await extensionRequestsQuery.fetchLatestExtensionRequest({
+        taskId: extensionBody.taskId,
+      });
+
+      if (latestExtensionRequest[0] && latestExtensionRequest[0].status === "PENDING") {
+        return res.boom.badRequest("An extension request for this task already exists.");
+      }
+
+      // dont allow newETA<taskEndsOnETA
+      if (latestExtensionRequest[0]) {
+        if (latestExtensionRequest[0].status === "APPROVED" || latestExtensionRequest[0].status === "DENIED") {
+          if (extensionBody.newEndsOn < task.endsOn) {
```

Comment:
There's a same check at line [45](https://github.com/Real-Dev-Squad/website-backend/pull/1563/files#diff-7fda274defc6e98dbba9ad405daf25c6db794428a333e30dc880bdfaa75ceb00R45), and it returns bad request. So this if-block will never be executed.

I think all three nested if-blocks can be removed here.

Reply:
Nice catch! I have removed that block of code. Thanks!